### PR TITLE
Enable running SGX workers as kubernetes pods.

### PR DESCRIPTION
**Pull request by manojgop**
Run enclave manager and workloads as kubernetes pod
in a system with SGX hardware support.

Signed-off-by: Manoj Gopalakrishnan <manoj.gopalakrishnan@intel.com>

### CODE REVIEW

File: docker/k8s/README.md [hunk 1]
```diff
@@ -7,14 +7,18 @@ in SGX-SIM mode.
...
+  `docker-compose -f docker-compose-sgx.yaml build`
 
 ## Running 
 
 1. Initialize the common configuration as a ConfigMap
     ```bash
     kubectl create -f tcf-config-map.yaml
     ```
-2. Enable `remote_url` in [tcs_config.toml](../../config/tcs_config.toml)
+2. Enable `remote_url` in [tcs_config.toml](../../config/tcs_config.toml)
```
Comment (danintel) [inline docker/k8s/README.md]: Remove the trailing whitespace you added to this line.

File: docker/k8s/README.md [hunk 2]
```diff
@@ -23,11 +27,17 @@ in SGX-SIM mode.
     # bootstrap the deployment and service
     kubectl create -f lmdb.yaml
     ```
-4. Start the `EnclaveManager` (to manage SGX enclaves) and `TCS Listener` (to
-   accept and handle requests from clients)  
+4. Start the `EnclaveManager` (to manage SGX enclaves)  
+   For SGX simulator mode, use below command
```
Comment (danintel) [inline docker/k8s/README.md]: s/use below command/use the command below/

File: docker/k8s/README.md (hunk 1 above)
Comment (manojgop) [reply docker/k8s/README.md]: two trailing spaces is new line in "mark down" document syntax.

File: docker/k8s/README.md (hunk 2 above)
Comment (manojgop) [reply docker/k8s/README.md]: Done

File: docker/k8s/sgx-enclave-manager-deployment.yaml [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+        - mountPath: /var/run/aesmd/aesm.socket
+          name: aesmsocket
+        securityContext:
+          privileged: true
+        envFrom:
+        - prefix: TCF_
+          configMapRef:
+            name: tcf-config
+        command: ["/bin/bash"]
+        args: ["$(TCF_HOME)/scripts/enclave_manager.sh"]
```
Comment (manju956) [inline docker/k8s/sgx-enclave-manager-deployment.yaml]: Is it a requirement to set TCF_HOME before running this yaml file? If so, can we mention this in documentation?

Comment (manojgop) [reply docker/k8s/sgx-enclave-manager-deployment.yaml]: This is set in https://github.com/hyperledger/avalon/blob/master/docker/k8s/tcf-config-map.yaml. Documentation mentions initializing the config map.

Comment (manju956) [reply docker/k8s/sgx-enclave-manager-deployment.yaml]: Okay

**Pull request merged**